testing/src/main/java/org/motechproject/nms/testing/service/impl/TestingServiceImpl.java

```diff
@@ -244,17 +257,37 @@ public Object execute(Query query) {
                 return null;
             }
         };
-        stateDataService.executeSQLQuery(sqe);
+        try {
+            stateDataService.executeSQLQuery(sqe);
+        } catch (Exception e) {
+            String s = String.format("Exception While executing \"%s\"", sqlQuery);
+            LOGGER.error(s);
+            throw e;
+        }
     }
 
 
-    private void disableConstraints() {
-        changeConstraints(true);
-    }
+    private void truncateTable(final String table) {
+        SqlQueryExecution sqe = new SqlQueryExecution() {
 
+            @Override
+            public String getSqlQuery() {
+                return String.format("DELETE FROM %s WHERE 1=1", table);
```

Comment:
That's what I originally thought, but...
Using delete clearDatabase() takes ~0.1 s
Using truncate, clearDatabase() takes ~4.0s

On Wed, Sep 2, 2015 at 10:52 AM, Rob LaRubbio <notifications@github.com>
wrote:

> In
> testing/src/main/java/org/motechproject/nms/testing/service/impl/TestingServiceImpl.java
> <https://github.com/motech-implementations/mim/pull/577#discussion_r38562760>
> :
>
> >
> > +            @Override
> > +            public String getSqlQuery() {
> > +                return String.format("DELETE FROM %s WHERE 1=1", table);
>
> Sorry I'm late to the party. Do you want to use delete or truncate? I
> think truncate would be faster.
>
> https://dev.mysql.com/doc/refman/5.0/en/truncate-table.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/577/files#r38562760>.
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201